pool/paymentmgr.go

```diff
@@ -644,13 +730,51 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32, treasuryA
 		return nil
 	}
 
+	if pm.isProcessing() {
+		log.Info("Waiting for in-progress payment processing to " +
```

Comment:
This isnt going to wait if `nil` is returned here. It will terminate the payment processing and stop it proceeding. Simply waiting for the mutex to be unlocked would fix this issue.

Reply:
I can see why you thought I wanted the call to wait and execute, will update the comment.